junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/predicates/IsTestableMethod.java

```diff
@@ -46,6 +49,10 @@ public boolean test(Method candidate) {
 		if (isAbstract(candidate)) {
 			return false;
 		}
+		if (returnsVoid(candidate) != mustReturnVoid) {
```

Comment:
hmmmm..... well, the "more readable version" seemed _logical_ to me, but if it's wrong... then it's wrong. 

Maybe I need to retake a basic course in boolean logic. 😉